src/irmin-pack/inode.ml

```diff
@@ -1098,6 +1087,35 @@ struct
       if depth = 0 then of_seq la (List.to_seq l)
       else values la (StepMap.of_list l)
 
+    let of_inode la ~depth ~length proofs =
+      let hash v = Bin.V.hash (to_bin_v la v) in
+      let new_entries () = Array.make Conf.entries None in
+      let rec aux entries proofs k =
+        match proofs with
+        | [] ->
+            let v = Tree { depth; length; entries } in
+            k { hash = lazy (hash v); stable = false; v }
+        | (index, hash) :: proofs -> (
+            match index with
+            | [] -> assert false
+            | [ index ] ->
+                let ptr = Ptr.of_hash la hash in
+                entries.(index) <- Some ptr;
+                aux entries proofs k
+            | index :: rest ->
+                let entries = new_entries () in
+                aux entries [ (rest, hash) ] (fun t ->
+                    let ptr = Ptr.of_target la t in
+                    entries.(index) <- Some ptr;
+                    aux entries proofs k))
+      in
+      aux (new_entries ()) proofs Fun.id
+
+    let of_elt la ~depth e =
+      match e with
+      | `Node n -> Some (of_values la ~depth n)
+      | `Inode (length, proofs) -> Some (of_inode la ~length ~depth proofs)
+
```

Comment:
Should this be stabilised?

Reply:
for inodes?

Reply:
Both

Reply:
to_elt will build a `Values` if it's a stable node so that should be ok. Should I add an assert to make sure `of_inode` is never called with `depth=0` and `lengh < Conf.stable_node` ?

Reply:
👍

Reply:
I've added an assert for `of_inode` ; `of_value` is calling `of_seq` at depth 0 which is calling `stabilize`